sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -32,7 +32,9 @@ object NestedColumnAliasing {
 
   def unapply(plan: LogicalPlan): Option[LogicalPlan] = plan match {
     case Project(projectList, child)
-        if SQLConf.get.nestedSchemaPruningEnabled && canProjectPushThrough(child) =>
+        if SQLConf.get.nestedSchemaPruningEnabled &&
+          (canProjectPushThrough(child) ||
+            getChild(child).exists(canProjectPushThrough)) =>
```

Comment:
Is this correct? I'm not 100% sure that this matching case can handle this condition: `Project->[*Any* logical unary node]->[Logical node that can be pushed through]`. Anyway, we need more tests for this change.

Reply:
Btw, I feel the title and the PR description are not accurate because this PR is not only for supporting the window case in nested pruning. Could you make them clearer?

Reply:
I don't think this is correct fix. It will push through a child that should not be pushed through.

Reply:
How about use my proposal?

Reply:
> That won’t work because it seems causing an infinite loop in optimizer. It gives me error messages like running out of max iterations.
>> I see, it is due to predicate pushdown rule.

I don't look into it though, we cannot fix the infinite loop caused by the predicate pushdown rule? If we can put `Filter` in `canProjectPushThrough`, it looks the best.

Reply:
> How about use my proposal at #28898 (review)?

If we cannot, yea, I think we need special handling for `Filter` as @viirya suggested above.

Reply:
Yeah, I will update this PR later tonight.

Reply:
I was trying to make this more generalized as I think `Project` may not be the only case I need to handle here. I just took a look at all the `UnaryNode` here, and it seems like only `Filter` needs special handling. I gonna adopt @viirya approach.